test/integration/auth.test.js

```diff
@@ -254,4 +256,191 @@ describe("auth", function () {
+    const res = await chai.request(app).get("/auth/google/login?dev=true").redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(googleOauthURL);
+  });
+
+  it("should return google call back URL with redirectUrl", async function () {
+    const RDS_MEMBERS_SITE_URL = "https://members.realdevsquad.com";
+    const googleOauthURL = generateGoogleAuthRedirectUrl({ state: RDS_MEMBERS_SITE_URL });
+    const res = await chai
+      .request(app)
+      .get("/auth/google/login?dev=true")
+      .query({ redirectURL: RDS_MEMBERS_SITE_URL })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(googleOauthURL);
+  });
+
+  it("should redirect the google user to new sign up flow if they are have incomplete user details true", async function () {
+    const redirectURL = "https://my.realdevsquad.com/new-signup";
+
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get("/auth/google/callback")
+      .query({ code: "codeReturnedByGoogle" })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(redirectURL);
+  });
+
+  it("should redirect the google user to the goto page on successful login, if user has incomplete user details false", async function () {
+    await addUserToDBForTest(googleUserInfo[1]);
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get("/auth/google/callback")
+      .query({ code: "codeReturnedByGoogle", state: rdsUiUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUiUrl);
+  });
+
+  it("should redirect the google user to the redirect URL provided on successful login, if user has incomplete user details false", async function () {
+    await addUserToDBForTest(googleUserInfo[1]);
+    const rdsUrl = new URL("https://dashboard.realdevsquad.com").href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get(`/auth/google/callback`)
+      .query({ code: "codeReturnedByGoogle", state: rdsUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUrl);
+  });
+
+  it("should redirect the google user to realdevsquad.com if non RDS URL provided, any url that is other than *.realdevsquad.com is invalid", async function () {
```

Comment:
No tests for duplicate email scenarios

Reply:
could you kindly clarify what type of test you are looking for regarding the "duplicate email scenarios"?

Reply:
For duplicate email scenarios, we should test:
- Same email trying to log in with different providers
- Existing email attempting new registration
- Email conflicts during user updates

Reply:
i have added test for this `Same email trying to log in with different providers`